test/fakeintake/aggregator/apmStatsAggregator.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package aggregator
+
+import (
+	"time"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+
+	"github.com/tinylib/msgp/msgp"
+)
+
+// APMStatsPayload is a payload type for APM stats
+// It implements PayloadItem
+type APMStatsPayload struct {
+	*pb.StatsPayload
+	collectedTime time.Time
+}
+
+var _ PayloadItem = &APMStatsPayload{}
+
+// name returns the hostname of the agent that produced the payload
+func (p *APMStatsPayload) name() string {
+	return p.AgentHostname
+}
+
+// GetTags is not implemented
+func (p *APMStatsPayload) GetTags() []string {
+	return nil
+}
+
+// GetCollectedTime returns the time that the payload was received by the fake intake
+// APMStatsPayload implements PayloadItem
+func (p *APMStatsPayload) GetCollectedTime() time.Time {
+	return p.collectedTime
+}
+
+// ParseAPMStatsPayload parses an api.Payload into a list of APMStatsPayload
+func ParseAPMStatsPayload(payload api.Payload) ([]*APMStatsPayload, error) {
+	rc, err := getReadCloserForEncoding(payload.Data, payload.Encoding)
+	if err != nil {
+		return nil, err
+	}
+	defer rc.Close()
+	var p pb.StatsPayload
+	if err := msgp.Decode(rc, &p); err != nil {
+		return nil, err
+	}
+	return []*APMStatsPayload{{StatsPayload: &p, collectedTime: payload.Timestamp}}, nil
```

Comment:
```suggestion
    stats := make([]*APMStatsPayload, 0, len(p.Stats)
    for _, stat := range p.Stats {
      stats = append(stats, &APMStatsPayload{StatsPayload: &stat, collectedTime: payload.Timestamp})
    }
	return stats, nil
```

This “unbundling” is what is done for the other types of payloads:
* https://github.com/DataDog/datadog-agent/blob/e3084595ca29763ec0784c6de691a1f8c1873a5b/test/fakeintake/aggregator/metricAggregator.go#L55-L60
* https://github.com/DataDog/datadog-agent/blob/e3084595ca29763ec0784c6de691a1f8c1873a5b/test/fakeintake/aggregator/containerLifecycleAggregator.go#L56-L60